From c9cc9e4c1c008fe70e442f0a8ad9b6abe9357270 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Mon, 11 Dec 2023 13:00:47 +0100
Subject: [PATCH 432/954] feat(stm32mp13-fdts): add nvram (NVMEM) node under
 tamp in stm32mp131.dtsi

Backup Registers nvram will be accessed using the NVMEM framework.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 279cfff9b39611fcd155be127f1ae5d9b8d5a929 feat(stm32mp13-fdts): add nvram (NVMEM) node under tamp in stm32mp131.dtsi]
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Change-Id: I512a5443621547500b9c3c4f7db79167570ca562
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/343946
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
---
 fdts/stm32mp131.dtsi | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/fdts/stm32mp131.dtsi b/fdts/stm32mp131.dtsi
index 2ed27374d9..3a800d0e1f 100644
--- a/fdts/stm32mp131.dtsi
+++ b/fdts/stm32mp131.dtsi
@@ -458,6 +458,31 @@
 				reg = <0x170 0x10>;
 			};
 		};
+
+		tamp: tamp@5c00a000 {
+			compatible = "st,stm32-tamp", "syscon", "simple-mfd";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x5c00a000 0x400>;
+			ranges;
+
+			nvram: nvram@5c00a100 {
+				compatible = "st,stm32mp15-tamp-nvram";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <0x5c00a100 0x80>;
+
+				fwu_info: tamp-bkp@28 {
+					/* see firmware update info feature */
+					reg = <0x28 0x4>;
+				};
+				boot_mode: tamp-bkp@78 {
+					/* see boot mode selection feature */
+					reg = <0x78 0x4>;
+				};
+			};
+		};
+
 		/*
 		 * Break node order to solve dependency probe issue between
 		 * pinctrl and exti.
-- 
2.53.0

